Subject: [PATCH] cgal 6 aab_traits removal ifdef
---
Index: src/ifcgeom/kernels/cgal/CgalKernel.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/ifcgeom/kernels/cgal/CgalKernel.cpp b/src/ifcgeom/kernels/cgal/CgalKernel.cpp
--- a/src/ifcgeom/kernels/cgal/CgalKernel.cpp	(revision 32899ebffbb1cfc9ff3e2325dd66a05c50b73483)
+++ b/src/ifcgeom/kernels/cgal/CgalKernel.cpp	(date 1728370210506)
@@ -1315,7 +1315,11 @@
 	if (proc == PP_SNAP_POINTS_TO_FIRST_OPERAND) {
 		static int NN = 0;
 		typedef CGAL::AABB_face_graph_triangle_primitive<cgal_shape_t>                AABB_face_graph_primitive;
-		typedef CGAL::AABB_traits<Kernel_, AABB_face_graph_primitive>               AABB_face_graph_traits;
+#if CGAL_VERSION_NR >= 1060000000
+	    typedef CGAL::AABB_traits_3<Kernel_, AABB_face_graph_primitive>               AABB_face_graph_traits;
+#else
+	    typedef CGAL::AABB_traits<Kernel_, AABB_face_graph_primitive>               AABB_face_graph_traits;
+#endif
 
 		CGAL::AABB_tree<AABB_face_graph_traits> tree;
 
