Subject: [PATCH] edit conda hdf5 packaging
---
Index: cmake/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
--- a/cmake/CMakeLists.txt	(revision 535e974cc334afd4f99abb820853ff49c5964b90)
+++ b/cmake/CMakeLists.txt	(date 1684823786831)
@@ -506,25 +506,29 @@
 
         ELSE()
             MESSAGE(STATUS "Packaging hdf5 and zlib for conda distribution")
+            set(HDF5_LIBRARIES)
             if (WIN32)
                 # Windows
-                set(zlib_post zlib)
-                set(lib_ext lib)
+                find_package(HDF5 COMPONENTS CXX)
+                if (HDF5_FOUND)
+                    set(HDF5_LIBRARIES ${HDF5_CXX_LIBRARIES})
+                endif()
+                list(APPEND HDF5_LIBRARIES
+                    "${HDF5_LIBRARY_DIR}/zlib.lib"
+                )
             elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
                 # macOS
-                set(zlib_post libz)
-                set(lib_ext dylib)
+                list(APPEND HDF5_LIBRARIES
+                    "${HDF5_LIBRARY_DIR}/libhdf5_cpp.dylib"
+                    "${HDF5_LIBRARY_DIR}/libhdf5.dylib"
+                    "${HDF5_LIBRARY_DIR}/libz.dylib"
+                )
             else()
                 # linux
-                set(zlib_post libz)
-                set(lib_ext so)
+                list(APPEND HDF5_LIBRARIES
+                    "${HDF5_LIBRARY_DIR}/libhdf5_cpp.so"
+                    "${HDF5_LIBRARY_DIR}/libhdf5.so"
+                    "${HDF5_LIBRARY_DIR}/libz.so"
+                )
             endif()
-
-            SET(HDF5_LIBRARIES
-                    "${HDF5_LIBRARY_DIR}/libhdf5_cpp.${lib_ext}"
-                    "${HDF5_LIBRARY_DIR}/libhdf5.${lib_ext}"
-                    "${HDF5_LIBRARY_DIR}/${zlib_post}.${lib_ext}"
-                    )
         endif()
 
     endif()
