Subject: [PATCH] fix boost mt19937 issue
---
Index: src/ifcparse/IfcGlobalId.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/ifcparse/IfcGlobalId.cpp b/src/ifcparse/IfcGlobalId.cpp
--- a/src/ifcparse/IfcGlobalId.cpp	(revision e072870fd38139a8fb9050fc21727fad580e459c)
+++ b/src/ifcparse/IfcGlobalId.cpp	(date 1725297302953)
@@ -88,7 +88,12 @@
 }
 
 // A random number generator for the UUID
+// Define the macro to handle different Boost versions
+#if BOOST_VERSION >= 108600
+static boost::uuids::basic_random_generator<std::mt19937> gen;
+#else
 static boost::uuids::basic_random_generator<boost::mt19937> gen;
+#endif
 
 IfcParse::IfcGlobalId::IfcGlobalId() {
     uuid_data_ = gen();
