Subject: [PATCH] suggested change of ifcwrap pythoninterpreter function to no longer rely on distutil (which is deprecated in python 3.12).
---
Index: src/ifcwrap/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/ifcwrap/CMakeLists.txt b/src/ifcwrap/CMakeLists.txt
--- a/src/ifcwrap/CMakeLists.txt	(revision c6643da4ba4afec799d205b0dfa85aaa8fe925f9)
+++ b/src/ifcwrap/CMakeLists.txt	(revision 0532a7005b9ca416953fee572f626d051342c61b)
@@ -78,7 +78,7 @@
             )
         ELSE ()
             EXECUTE_PROCESS(
-                COMMAND ${PYTHON_EXECUTABLE} -c "import sys; from distutils.sysconfig import get_python_lib; sys.stdout.write(get_python_lib(1))"
+                COMMAND ${PYTHON_EXECUTABLE} -c "import sys; import sysconfig; sys.stdout.write(sysconfig.get_path('platlib'))"
                 OUTPUT_VARIABLE python_package_dir
             )
         ENDIF()
